elastalert/elastalert.py

```diff
@@ -1107,6 +1170,11 @@ def find_pending_aggregate_alert(self, rule):
 
     def add_aggregated_alert(self, match, rule):
```

Comment:
What if get_aggregated_matches and send_pending_alerts were left (almost) verbatim. And the following changes were added into add_aggregated_alert:

rule['current_aggregate_id'] becomes rule['current_aggregate_id'].get(query_key_value)
rule['aggregate_alert_time'] becomes rule['aggregate_alert_time'].get(query_key_value)

and the relevant code is updated in other places to initialize this object to a dictionary.